src/hotspot/share/opto/superword.cpp

```diff
@@ -3177,34 +2996,6 @@ BasicType SuperWord::longer_type_for_conversion(Node* n) const {
                               : (src_size > dst_size ? src_t : dst_t);
 }
 
-int SuperWord::max_vector_size_in_def_use_chain(Node* n) {
```

Comment:
Note: was used by `get_vw_bytes_special`. It looks at inputs and outputs of the node `n`, and looks for the largest basic type via `longer_type_for_conversion`. It then returned the max vector size (i.e. number of elements) for that basic type. We can fit fewer large elements in a vector. If we have small elements, we would like to have many elements in a vector. But we must make sure that use and def vectors can have at least as many elements.

After I had recently introduced `split_packs_at_use_def_boundaries`, this special logic here is no longer necessary.